edb/schema/scalars.py

```diff
@@ -269,20 +274,28 @@ def _cmd_tree_from_ast(
             create_cmd = cmd
 
         bases = create_cmd.get_attribute_value('bases')
-        is_enum = False
-        if len(bases) == 1 and isinstance(bases._ids[0], AnonymousEnumTypeRef):
-            # type ignore below because this class elements is set
-            # directly on __dict__
-            elements = bases._ids[0].elements  # type: ignore
+        is_enum = (
+            len(bases) == 1
+            and bases.first(schema).get_name(schema) == 'std::anyenum'
+        )
```

Comment:
What if there are more bases and one of them is an enum? The variable name `is_enum` is confusing; add a comment?

Reply:
This is rewritten in a subsequent commit.  At this particular point the check is done in `_validate_base_refs()` earlier.